pkg/manifests/config.go

```diff
@@ -126,6 +127,27 @@ func (c Config) GetThanosRulerAlertmanagerConfigs() []AdditionalAlertmanagerConf
 	return alertmanagerConfigs
 }
 
+func (c Config) HasInvalidScrapeIntervalDuration() bool {
```

Comment:
It'd be better to return an error that would be wrapped by the caller with `ErrUserWorkloadValidation`.
```suggestion
func (c Config) checkUserWorkloadPrometheusConfig() error {
```